syncd/Syncd.cpp

```diff
@@ -898,6 +959,122 @@ sai_status_t Syncd::processEntry(
     }
 }
 
+sai_status_t Syncd::processBulkOidCreate(_In_ sai_object_type_t objectType,
+            _In_ sai_bulk_op_error_mode_t mode,
+            _In_ const std::vector<std::string>& objectIds,
+            _In_ const std::vector<std::shared_ptr<saimeta::SaiAttributeList>>& attributes,
+            _Out_ std::vector<sai_status_t>& statuses)
+{
+    SWSS_LOG_ENTER();
+
+    /*
+     * Switch, Port objects doesn't have bulk API`s, so we no need additional
+     * functionality for process it
+     */
+    if (objectType == SAI_OBJECT_TYPE_SWITCH || objectType == SAI_OBJECT_TYPE_PORT)
+    {
+        return SAI_STATUS_NOT_SUPPORTED;
+    }
+
+    sai_status_t status;
+
+    std::vector<sai_object_id_t> objectVids(objectIds.size());
+
+    std::vector<uint32_t> attr_counts(objectIds.size());
+    std::vector<const sai_attribute_t*> attr_lists(objectIds.size());
+
+    for (size_t idx = 0; idx < objectIds.size(); idx++)
+    {
+        sai_deserialize_object_id(objectIds[idx], objectVids[idx]);
+
+        attr_counts[idx] = attributes[idx]->get_attr_count();
+        attr_lists[idx] = attributes[idx]->get_attr_list();
+    }
+
+    sai_object_id_t switchRid = SAI_NULL_OBJECT_ID;
+
+    if (objectVids.size() > 0 && objectType != SAI_OBJECT_TYPE_SWITCH)
+    {
+        sai_object_id_t switchVid = VidManager::switchIdQuery(objectVids.front());
+        switchRid = m_translator->translateVidToRid(switchVid);
+    }
+
+    std::vector<sai_object_id_t> objectRids(objectIds.size());
+
+    status = m_vendorSai->bulkCreate(objectType, switchRid, (uint32_t)objectIds.size(), attr_counts.data(),
+                                        attr_lists.data(), mode, objectRids.data(), statuses.data());
+
+    if (status == SAI_STATUS_SUCCESS)
```

Comment:
this check is invalid, according to bulk API, success will be returned only when all create will succeed, if any fails, then return will be FAILURE, but even do general status is FAILURE, lest say 99 of 100 could be succeeded, and then you need to check statuses vector for each object to figure out which one failed, and insert to translator only those with success, and skip those without